### ISSUE TITLE
Support Hyperelliptic curves of genus >2

### ISSUE THREAD

**Issue by koffie** [wishcow79/chabauty#13]
Currently one has:

```
derickx@btm2x13:~/code/chabauty$ magma
Benutze Version avx64 static
Magma V2.23-4     Thu Nov  2 2017 17:55:08 on btm2x13  [Seed = 1858894587]
Type ? for help.  Type <Ctrl>-D to quit.
> load "chabauty.m";
Loading "chabauty.m"
> P<t> := PolynomialRing(Rationals());
> f := t*(t^7-2*t^5+16*t^4-15*t^3+36*t^2-32*t+4);
> C := HyperellipticCurve(f);
> ChabautyColeman(C);
Computing genus of curve.
I am timing this:
Time: 0.000
attempting Chabauty Coleman on curve of genus 3

Searching for rational points up to height 100000:
I am timing this:

ChabautyColeman(
    C: S
)
In file "chabauty.m", line 1223, column 28:
>>     time pts := PointSearch(C, HeightBound);
                              ^
Runtime error in 'PointSearch': Scheme cannot be in weighted proj space
```

**Comment by koffie** [comment]
Ok, this error was trivially to get past. Then there was another issue in ReduceCurveModp which was also easy to fix, but now I am stuck at:
```
> ChabautyColeman(C);
Computing genus of curve.
I am timing this:
Time: 0.000
attempting Chabauty Coleman on curve of genus 3

Searching for rational points up to height 100000:
I am timing this:
There are 3 rational points on C.
the points found are: 

P_1 = [ 1, -1, 0 ]
P_2 = [ 1, 1, 0 ]
P_3 = [ 0, 0, 1 ]
Using the following good primes for the algorithm:[ 3, 5, 7, 11, 13 ]
Torsion bounds for Jacobian: [ 3 ]
Lower bound on rank of the Jacobian subgroup: 1
Upper bound on rank of the Jacobian subgroup: 1
Upper bound = lower bound, so we can proceed.
Good prime chosen for Chabauty-Coleman: 3
Reduction mod 3 is surjective: false 
Precision sufficient to calculate integrals to precision O(p^5) is: 13
I chose base point P_1 = (1 : -1 : 0) (reduces to non-Weierstrass point mod 3)

ChabautyColeman(
    C: Hyperelliptic Curve defined by y^2 = x^8 - 2*x^6 + 16*x^5 - ...
)
GetGoodUniformizer(
    basept: (1 : -1 : 0),
    p: 3
)
Valuation(
    f: ($.1 + $.2)/$.2,
    p: (1 : 2 : 0)
)
In file "/programs/magma/magma.2.23-4/package/Geometry/Crv/fld_fun.m", line 146,
column 21:
>>     return Valuation(f,pls[1]);
                       ^
Runtime error in 'Valuation': First argument is not a function on the scheme 
containing the second argument
```
I'll upload the previous mentioned fixes hopefully tomorrow.

**Comment by koffie** [comment]
Ok, I added the two fixes at 8d71a3d4819757835d641f240b45f3da0c04d16e but as mentioned there is still stuff to be done for hyperelliptic curves

**Comment by koffie** [comment]
This was fixed by #19
